### PR TITLE
`azurerm_application_gateway` - support for `verify_client_certificate_revocation` in `ssl_profile` block

### PR DESCRIPTION
**Pull request by teowa**
- [Certificate Revocation doc](https://learn.microsoft.com/en-us/azure/application-gateway/mutual-authentication-overview?tabs=powershell#certificate-revocation)
- [REST API](https://github.com/Azure/azure-rest-api-specs/blob/c320f58d8643a309c126dd4ec5fdc6737fccd981/specification/network/resource-manager/Microsoft.Network/stable/2022-07-01/applicationGateway.json#L1371)

### CODE REVIEW

File: internal/services/network/application_gateway_resource.go [hunk 1]
```diff
@@ -1245,6 +1246,16 @@ func resourceApplicationGateway() *pluginsdk.Resource {
 							Default:  false,
 						},
 
+						"verify_client_cert_revocation": {
```
Comment (stephybun) [inline internal/services/network/application_gateway_resource.go:1249]: We should spell out certificate here - I see that the above property has it abbreviated, this should be updated in 4.0 please add a comment above it to that effect
```suggestion
						"verify_client_certificate_revocation": {
```

File: internal/services/network/application_gateway_resource.go [hunk 2]
```diff
@@ -1245,6 +1246,16 @@ func resourceApplicationGateway() *pluginsdk.Resource {
...
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							Default:  string(network.ApplicationGatewayClientRevocationOptionsNone),
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.ApplicationGatewayClientRevocationOptionsOCSP),
+								string(network.ApplicationGatewayClientRevocationOptionsNone),
```
Comment (stephybun) [inline internal/services/network/application_gateway_resource.go:1255]: We don't expose values like `None`

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @teowa.

We don't expose values like `None` as explained in [this section of the contributor docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#the-none-value-or-similar). Can you please update the schema as well as the property name so that it follows our [naming guidelines](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-naming.md#property-naming)

**Comment by teowa** [comment]
image

File: internal/services/network/application_gateway_resource.go [hunk 3]
```diff
@@ -4218,11 +4228,15 @@ func expandApplicationGatewaySslProfiles(d *pluginsdk.ResourceData, gatewayID st
...
+		verifyClientCertificateRevocation := v["verify_client_certificate_revocation"].(string)
 
 		output := network.ApplicationGatewaySslProfile{
-			Name: utils.String(name),
+			Name: pointer.To(name),
 			ApplicationGatewaySslProfilePropertiesFormat: &network.ApplicationGatewaySslProfilePropertiesFormat{
-				ClientAuthConfiguration: &network.ApplicationGatewayClientAuthConfiguration{VerifyClientCertIssuerDN: utils.Bool(verifyClientCertIssuerDn)},
+				ClientAuthConfiguration: &network.ApplicationGatewayClientAuthConfiguration{
+					VerifyClientCertIssuerDN: pointer.To(verifyClientCertIssuerDn),
+					VerifyClientRevocation:   network.ApplicationGatewayClientRevocationOptions(verifyClientCertificateRevocation),
```
Comment (stephybun) [inline internal/services/network/application_gateway_resource.go:4238]: If no value is provided for `verify_client_certificate_revocation` then we will be sending an empty string here? Shouldn't we be defaulting this to `None` if this isn't set?

Comment (teowa) [reply internal/services/network/application_gateway_resource.go:4238]: behaviorally it is same, the API return `None` even if it is not set.

**Review by stephybun — APPROVED** [review]
Thanks @teowa LGTM 🎸

**Pull request merged**